src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java

```diff
@@ -211,11 +215,24 @@ public void verifyAgencyAccess(final String agencyId) {
         if (AuthenticationFacade.hasRoles("INACTIVE_BOOKINGS")) {
             agencyIds.addAll(Set.of("OUT", "TRN"));
         }
+        if (agencyIds.isEmpty()) {
+            if (authenticationFacade.isClientOnly()) {
+                logClientUnauthorisedAccess(agencyId);
+            }
+            throw EntityNotFoundException.withId(agencyId);
```

Comment:
This is similar to the one for bookings.  The overrideRoles is not passed into this method, but if we need it - we can easily pass this through for logging